frontend/server/src/main/java/com/amazonaws/ml/mms/http/HttpRequestHandler.java

```diff
@@ -75,6 +88,72 @@ protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest req) {
         }
     }
 
+    private void run(
+            ModelServerEndpoint endpoint,
+            FullHttpRequest req,
+            FullHttpResponse rsp,
+            QueryStringDecoder decoder,
+            String method)
+            throws IOException {
+        switch (method) {
+            case "GET":
+                endpoint.doGet(
+                        new ModelServerRequest(req, decoder),
+                        new ModelServerResponse(rsp),
+                        new ModelServerContext());
+                break;
+            case "PUT":
+                endpoint.doPut(
+                        new ModelServerRequest(req, decoder),
+                        new ModelServerResponse(rsp),
+                        new ModelServerContext());
+                break;
+            case "DELETE":
+                endpoint.doDelete(
+                        new ModelServerRequest(req, decoder),
+                        new ModelServerResponse(rsp),
+                        new ModelServerContext());
+                break;
+            case "POST":
+                endpoint.doPost(
+                        new ModelServerRequest(req, decoder),
+                        new ModelServerResponse(rsp),
+                        new ModelServerContext());
+                break;
+            default:
+                throw new ServiceUnavailableException("Invalid HTTP method received");
+        }
+    }
+
+    protected void handleCustomEndpoint(
+            ChannelHandlerContext ctx,
+            FullHttpRequest req,
+            String[] segments,
+            QueryStringDecoder decoder) {
+        ModelServerEndpoint endpoint = endpointMap.get(segments[1]);
+        Runnable r =
+                () -> {
+                    Long start = System.currentTimeMillis();
+                    FullHttpResponse rsp =
+                            new DefaultFullHttpResponse(
+                                    HttpVersion.HTTP_1_1, HttpResponseStatus.OK, false);
+                    try {
+                        run(endpoint, req, rsp, decoder, req.method().toString());
+                        NettyUtils.sendHttpResponse(ctx, rsp, true);
+                        logger.info(
+                                "Running \"{}\" endpoint took {} ms",
+                                segments[0],
+                                System.currentTimeMillis() - start);
+                    } catch (Exception e) {
+                        ByteArrayOutputStream ps = new ByteArrayOutputStream();
```

Comment:
Send stracktrace back to client isn't a good idea. This might be a security issue.

Reply:
Ya. Agreed. I removed it. I am just logging it in logs for now.